app/src/main/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagsViewModel.kt

```diff
@@ -20,42 +23,99 @@ import javax.inject.Inject
 class FeatureFlagsViewModel @Inject constructor(
   private val platformParameterControllerDebugImpl: PlatformParameterControllerDebugImpl,
   private val machineLocale: OppiaLocale.MachineLocale,
-  private val resourceHandler: AppLanguageResourceHandler
+  private val resourceHandler: AppLanguageResourceHandler,
 ) : ObservableViewModel() {
+  /**
+   * LiveData that contains a list of [FeatureFlagItemViewModel] which is used to populate the
+   * recycler view in [FeatureFlagsFragment].
+   */
+  val featureFlagList: LiveData<List<FeatureFlagItemViewModel>> by lazy {
+    Transformations.map(ephemeralFlagsLiveData, ::processFeatureFlagList)
+  }
+
   private val ephemeralFlagsLiveData: LiveData<List<EphemeralFeatureFlag>> by lazy {
     Transformations.map(
       platformParameterControllerDebugImpl.loadEphemeralFeatureFlags().toLiveData(),
       ::processEphemeralFlagResult
     )
   }
 
+  /** List of feature flag switch states to be used in the fragment. */
+  val featureFlagStates = MutableLiveData<MutableMap<FeatureFlagId, Boolean>>(mutableMapOf())
+
+  /** List of feature flags that have been reset. */
+  val resetFlags = MutableLiveData<MutableMap<FeatureFlagId, Boolean>>(mutableMapOf())
+
+  /** Tracks whether the Save button is currently enabled (clickable). */
+  val isSaveButtonEnabled: LiveData<Boolean> by lazy {
+    Transformations.map(featureFlagStates) { it.isNotEmpty() }
+  }
+
   /**
-   * LiveData that contains a list of [FeatureFlagItemViewModel] which is used to populate the
-   * recycler view in [FeatureFlagsFragment].
+   * Updates the state of a given feature flag with a new value.
+   * This method handles the logic for determining whether to add, update, or remove
+   * the flag from the tracked states based on the original value and reset status.
+   *
+   * @param id the [FeatureFlagId] of the feature flag being updated.
+   * @param newValue the new value for the feature flag.
+   * @param originalValue the original (non-overridden) value of the feature flag.
    */
-  val featureFlagList: LiveData<List<FeatureFlagItemViewModel>> by lazy {
-    Transformations.map(ephemeralFlagsLiveData, ::processFeatureFlagList)
+  fun updateFeatureFlagState(
+    id: FeatureFlagId,
+    newValue: Boolean,
+    originalValue: Boolean
+  ) {
+    val currentStates = featureFlagStates.value?.toMutableMap() ?: mutableMapOf()
+    val currentResetFlags = resetFlags.value ?: mutableMapOf()
+    if (originalValue == newValue && !currentResetFlags.containsKey(id)) {
```

Comment:
Can you please use the same suggestion as before: https://github.com/oppia/oppia-android/pull/5894#discussion_r2292536621
```suggestion
    if (originalValue == newValue && id !in currentResetFlags) {
```

Also, I can see the exact function written in a different format with `PlatformParametersViewModel`, please follow a consistent way throughout.

Reply:
updated to maintain consistency